pkg/asset/agent/installconfig.go

```diff
@@ -120,3 +124,172 @@ func (a *OptionalInstallConfig) ClusterName() string {
+				logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, host.BMC.Password))
+			}
+			if host.BMC.Address != "" {
+				fieldPath := field.NewPath("Platform", "Baremetal", fmt.Sprintf("Hosts[%d]", i), "BMC", "Address")
+				logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, host.BMC.Address))
+			}
+			if host.BMC.DisableCertificateVerification {
+				fieldPath := field.NewPath("Platform", "Baremetal", fmt.Sprintf("Hosts[%d]", i), "BMC", "DisableCertificateVerification")
+				logrus.Warnf(fmt.Sprintf("%s: true is ignored", fieldPath))
+			}
+			if host.Role != "" {
+				fieldPath := field.NewPath("Platform", "Baremetal", fmt.Sprintf("Hosts[%d]", i), "Role")
+				logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, host.Role))
+			}
+			if host.HardwareProfile != "" {
+				fieldPath := field.NewPath("Platform", "Baremetal", fmt.Sprintf("Hosts[%d]", i), "HardwareProfile")
+				logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, host.HardwareProfile))
+			}
+			if host.RootDeviceHints != nil {
+				fieldPath := field.NewPath("Platform", "Baremetal", fmt.Sprintf("Hosts[%d]", i), "RootDeviceHints")
+				logrus.Warnf(fmt.Sprintf("%s is ignored", fieldPath))
+			}
+			// The default is UEFI. +kubebuilder:validation:Enum="";UEFI;UEFISecureBoot;legacy. Set from generic install config code
+			if host.BootMode != "UEFI" {
+				fieldPath := field.NewPath("Platform", "Baremetal", fmt.Sprintf("Hosts[%d]", i), "BootMode")
+				logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, host.BootMode))
+			}
+			if host.NetworkConfig != nil {
+				fieldPath := field.NewPath("Platform", "Baremetal", fmt.Sprintf("Hosts[%d]", i), "NetworkConfig")
+				logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, host.NetworkConfig))
+			}
+		}
+
+		if baremetal.DefaultMachinePlatform != nil {
+			fieldPath := field.NewPath("Platform", "Baremetal", "DefaultMachinePlatform")
+			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.DefaultMachinePlatform))
+		}
+		if baremetal.BootstrapOSImage != "" {
+			fieldPath := field.NewPath("Platform", "Baremetal", "BootstrapOSImage")
+			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.BootstrapOSImage))
+		}
+		if baremetal.ClusterOSImage != "" {
+			fieldPath := field.NewPath("Platform", "Baremetal", "ClusterOSImage")
+			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.ClusterOSImage))
+		}
+		if baremetal.BootstrapExternalStaticIP != "" {
+			fieldPath := field.NewPath("Platform", "Baremetal", "BootstrapExternalStaticIP")
+			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.BootstrapExternalStaticIP))
+		}
+		if baremetal.BootstrapExternalStaticGateway != "" {
+			fieldPath := field.NewPath("Platform", "Baremetal", "BootstrapExternalStaticGateway")
+			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.BootstrapExternalStaticGateway))
+		}
+	}
```

Comment:
We also ignore the following fields in Platform.VSphere:

* VCenter
* Username
* Password
* Datacenter
* DefaultDatastore
* Folder
* Cluster
* ResourcePool
* ClusterOSImage
* DefaultMachinePlatform
* Network
* DIskType
* VCenters
* FailureDomains

Reply:
Done